script.js

```diff
@@ -389,7 +389,8 @@ window.GaiaAppIcon = (function(exports) {
   };
 
   proto._relayout = function() {
-    this._size = this.clientWidth;
+    var bookmarkIcon = this.bookmark && this.bookmark.icon;
+    this._size = this.clientWidth || (bookmarkIcon && bookmarkIcon.size);
```

Comment:
The problem is that we are not appending the element to the DOM in the case of the chrome. We just replace the backgroundImage with the one created by the component. I tried to set the size via style without adding it to the DOM and didn't seem to work for me...